### PR TITLE
[framework] dont clip navigator by default.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Part of https://github.com/flutter/flutter/issues/136510

The navigator clips by themself are not slow, but they are disabling clear color optimizations. The clear color optimization detects that a draw's geometry full covers the surface that is being rendered to. Clipping disables this as we can't be sure if the clip impacts the result. Because most route backgrounds are implemented as a large drawRect or drawPaint, losing clear color adds about 3-4ms of GPU time.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/navigator.dart [hunk 1]
```diff
@@ -1461,7 +1461,7 @@ class Navigator extends StatefulWidget {
     this.onUnknownRoute,
     this.transitionDelegate = const DefaultTransitionDelegate<dynamic>(),
     this.reportsRouteUpdateToEngine = false,
-    this.clipBehavior = Clip.hardEdge,
```
Comment (chunhtai) [inline packages/flutter/lib/src/widgets/navigator.dart:1464]: If this is a nested Navigator on only part of the screen, any route transition that does paint transform (MaterialPageRoute for example) will be drawn outside of the rect.

I think for most nested navigation scenario you will want this to clip.

**Review by chunhtai — COMMENTED** [review]
Does this really help? I think using nested navigator is not that common

File: packages/flutter/lib/src/widgets/navigator.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter/lib/src/widgets/navigator.dart:1464]: Isn't this the per overlay clip? Or did I totally misunderstand this code

Comment (jonahwilliams) [reply packages/flutter/lib/src/widgets/navigator.dart:1464]: If its just nested navigators then ignore this, I misread the code!

**Pull request closed without merging**